### ISSUE TITLE
Flag for sourmash compare to change sample names

### ISSUE THREAD

**Issue by halexand** [dib-lab/q2-sourmash#2]
The integration of the output DistanceMatrix with subsequent QIIME2 tools (e.g. Emperor) would be better facilitated if the naming of the samples in the DistanceMatrix output by `sourmash compare` could be made to correspond with Sample Metadata tables. 

For example with the Moving Pictures Tutorial the sample data table (sample_metadata.tsv): 

> #SampleID	BarcodeSequence	LinkerPrimerSequence	BodySite	Year	Month	Day	Subject	ReportedAntibioticUsage	DaysSinceExperimentStart	Description
#q2:types	categorical	categorical	categorical	numeric	numericnumeric	categorical	categorical	numeric	categorical
L1S8	AGCTGACTAGTC	GTGCCAGCMGCCGCGGTAA	gut	2008	10	28	subject-1	Yes	0	subject-1.gut.2008-10-28
L1S57	ACACACTATGGC	GTGCCAGCMGCCGCGGTAA	gut	2009	1	20	subject-1	No	84	subject-1.gut.2009-1-20
L1S76	ACTACGTGTGGT	GTGCCAGCMGCCGCGGTAA	gut	2009	2	17	subject-1	No	112	subject-1.gut.2009-2-17

But the distance matrix output by sourmash is listed by the file paths: 

> /var/folders/r9/86xmxscs2fx26664gnv0wk3h0000gn/T/q2-MinHashSigJsonDirFormat-gkquzqjf/L1S76_12_L001_R1_001.fastq.gz      0.9375  0.8333333333333334      0.8235294117647058      0.8571428571428572      0.8666666666666667      0.6363636363636364      0.0     0.8666666666666667      1.0     1.0     0.9583333333333334      0.9565217391304348      1.0     1.0     0.9333333333333333      1.0     0.9090909090909091      1.0     1.0     0.9333333333333333      1.0     0.8     0.9722222222222222      1.0     0.9722222222222222      1.0     1.0     1.0     1.0     1.0     1.0     1.0     1.0     1.0
/var/folders/r9/86xmxscs2fx26664gnv0wk3h0000gn/T/q2-MinHashSigJsonDirFormat-gkquzqjf/L1S8_8_L001_R1_001.fastq.gz        0.9411764705882353      0.9285714285714286      0.8333333333333334      0.8666666666666667      0.9411764705882353      0.8571428571428572      0.8666666666666667      0.0     1.0     1.0     1.0     1.0     1.0     1.0     1.0     1.0     0.9166666666666666      1.0     1.0     1.0     1.0     0.9166666666666666      1.0     1.0     1.0     1.0     1.0     1.0     1.0     1.0     1.0     1.0     1.0     1.0

**Comment by halexand** [comment]
Any thoughts on a straightforward fix @luizirber @ctb @jiarong?

**Comment by jiarong** [comment]
Sourmash take sequence file name as id.  But the sample id to sequence file name mapping is not in the sample metadata table, so there is no robust way. A hacky fix is to parse the sample id out of the file name.

**Comment by halexand** [comment]
Yeah, didn't think that would be good as the file names do not map onto the sample names in a consistent fashion.

**Comment by luizirber** [comment]
The information is not in the sample metadata table, but is it somewhere else?

**Comment by jiarong** [comment]
I do not think there is any. We have to guess the SampleID in the metadata from the file names.

**Comment by halexand** [comment]
I think I might have found a solution. When Fastq data is loaded into QIIME the sample metadata is required-- it generates a file called `MANIFEST` that is links sample name (used down stream) to the fastq file names. We copy the fastq files into what becomes the output folder as part of our running of sourmash in QIIME-- so I have just made it so we scrape the linkage info from `MANIFEST` and then copy the files to the sample-id.fastq.gz. It is a little hacky-- but it should work somewhat broadly? Will post the PR later tonight.

**Comment by jiarong** [comment]
Nice solution with the MANIFEST file. I think it should work broadly.
